openedx/stanford/lms/envs/aws.py

```diff
@@ -34,6 +34,8 @@
 )
 PAYMENT_PLATFORM_NAME = ENV_TOKENS.get('PAYMENT_PLATFORM_NAME', PAYMENT_PLATFORM_NAME)
 PAYMENT_SUPPORT_PHONE = ENV_TOKENS.get('PAYMENT_SUPPORT_PHONE', PAYMENT_SUPPORT_PHONE)
+SEARCH_DISCOVERY_SORT_FIELDS = '_score:desc,start:desc'
+SEARCH_FILTER_GENERATOR = 'openedx.stanford.lms.lib.courseware_search.lms_filter_generator.TileSearchFilterGenerator'
```

Comment:
Is this the preferred way to avoid running it for tests, or could we just unset them in `openedx/stanford/lms/envs/test.py`?
```
SEARCH_DISCOVERY_SORT_FIELDS = ''
SEARCH_FILTER_GENERATOR = None
```

Reply:
I'm not sure if there's a universally preferred way.
I think we could do either. This way is a little less verbose;
just setting it once (`aws.py`), instead of twice (+`test.py`),
ableit not how we normally do it. 

Semantically, I think it's the difference between:
- only set these values if we're actually running a server [1] 
- set these values universally, override to empty values during tests

It appears that the former (the current implementation) provides an
identical settings object to what upstream would provide. It may not 
matter (now), but it's possible for the `edx-search` library to behave
differently whether a value is "set, at all" or "set to default values".
Think of it in JavaScript terms as `undefined` vs `null`.

```py
if hasattr(settings, 'SEARCH_DISCOVERY_SORT_FIELDS'):
    sort = settings.SEARCH_DISCOVERY_SORT_FIELDS
else:
    # where this is some `edx-search`-provided default that may change
    sort = '_score:asc,start:asc'
```

Defensively, I'm inclined to stick with what we have, to more closely
mirror upstream for now.


Thoughts?


- [1] diagram settings "inheritance"
    ```
    /tmp/django-settings
    |-- common.py
        |-- aws.py  --> production.py
            |-- devstack.py
            |-- test.py
    ```

Reply:
@kluo I updated the diagram; I mistakenly listed `test.py` as a child of `aws.py`.